src/App.jsx

```diff
@@ -1,41 +1,70 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 
-const App = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
+const App = () => {
+  const [userID, setuserID] = useState(0);
+  const [postID, setPostID] = useState(null);
+  const [posts, setPosts] = useState([]);
+  const [comments, setComments] = useState([]);
```

Comment:
You're passing empty comments and empty posts to your children components.
It's better to create these variables there. E.g. in PostDetails you ned only posted, with which you can fetch THERE everything you need (post details and comments)
Also, you can fetch here on componentDidMoun (using hooks) needed posts and pass them to 'PostsList', instead of passing empty posts